### PR TITLE
View record fields in the debugger

### PR DESCRIPTION
**Pull request by stevemessick**
This also includes a patch to `BoundField` that makes it usable, but I expect that will change with the next update to VM service protocol.

@alexander-doroshko

### CODE REVIEW

File: flutter-idea/third_party/vmServiceDrivers/org/dartlang/vm/service/element/BoundField.java [hunk 1]
```diff
@@ -42,11 +42,17 @@ public FieldRef getDecl() {
...
   public Object getName() {
-    final JsonObject elem = (JsonObject)json.get("name");
+    final JsonElement elem = json.get("name");
     if (elem == null) return null;
 
-    if (elem.get("type").getAsString().equals("String")) return elem.getAsString();
-    if (elem.get("type").getAsString().equals("int")) return elem.getAsInt();
+    if (elem.isJsonObject()) return elem.getAsString();
+    if (elem.isJsonPrimitive()) {
+      try {
```
Comment (jacob314) [inline flutter-idea/third_party/vmServiceDrivers/org/dartlang/vm/service/element/BoundField.java:50]: Nit: avoid using try-catch for non-exceptional cases where possible.

**Review by jacob314 — APPROVED** [review]
lgtm

**Comment by stevemessick** [comment]
https://github.com/dart-lang/sdk/issues/51514

File: flutter-idea/third_party/vmServiceDrivers/org/dartlang/vm/service/element/BoundField.java [hunk 2]
```diff
@@ -42,11 +42,17 @@ public FieldRef getDecl() {
    * @return one of <code>String</code> or <code>int</code>
    */
   public Object getName() {
-    final JsonObject elem = (JsonObject)json.get("name");
+    final JsonElement elem = json.get("name");
```
Comment (devoncarew) [inline flutter-idea/third_party/vmServiceDrivers/org/dartlang/vm/service/element/BoundField.java:45]: FWIW, this system of record for this file is in https://github.com/dart-lang/sdk/tree/main/pkg/vm_service/java; that lib is generated from the service protocol spec. Any changes here - which aren't upstreamed there - will be written over the next time these files are updated.

Comment (stevemessick) [reply flutter-idea/third_party/vmServiceDrivers/org/dartlang/vm/service/element/BoundField.java:45]: I've updated the VM service protocol twice recently, and expect to do so again now that the problems have been addressed.
https://github.com/dart-lang/sdk/issues/51448
https://github.com/dart-lang/sdk/issues/51514

**Pull request merged**